mambaSharedFramework/HLSValueTypes.swift

```diff
@@ -130,6 +131,92 @@ public func ==(lhs: HLSEncryptionMethodType, rhs: HLSEncryptionMethodType) -> Bo
     return lhs.type == rhs.type
 }
 
+/// Represents a minimum required HDCP level needed to play content.
+public struct HLSHDCPLevel: Equatable, FailableStringLiteralConvertible {
+    public let type: HDCPLevel
+    public enum HDCPLevel: String {
+        /// Indicates that the content does not require output copy protections.
+        case none = "NONE"
+        /// Indicates that the Variant Stream could fail to play unless the output is protected by High-bandwidth
+        /// Digital Content Protection (HDCP) Type 0 or equivalent.
+        case type0 = "TYPE-0"
+        /// Indicates that the Variant Stream could fail to play unless the output is protected by HDCP Type 1 or
+        /// equivalent.
+        case type1 = "TYPE-1"
+    }
+    public init?(string: String) {
+        self.init(hdcpLevel: string)
+    }
+    public init?(hdcpLevel: String) {
```

Comment:
I know the previous init is required for the protocol but is this one required? Is the purpose just that it reads nicer?

Reply:
Again, I'm really not sure why this style is taken, I was just following the style of the rest of the file... I'm happy to make a change where I favor a more enum-centric approach. I'll look to make a change in that direction and see how it goes.

Reply:
Updated here: 34de4a993d2206f793afe18e53db83b07d7f8ca8